### PR TITLE
feat: match exercise order to book chapters

### PR DESCRIPTION
**Pull request by JuliaCao**
Exercises are not organized in order for ```rustlings watch``` and it was hard to follow the exercises because we always had to jump back and forth between chapters.

1. Updated exercise order to match that of book chapters
2. Added exercise to book chapter mapping table to exercise README.md

### CODE REVIEW

**Comment by darnuria** [comment]
This pr need some local testing? If so I can checkout and test myself I use rustling with students and feel some problems due to mismatch with book chapters/rustlings order.

**Comment by JuliaCao** [comment]
Hello,

I have tested this locally. But feel free to double check :)

**Comment by JuliaCao** [comment]
> Hi, this pr need some local testing? If so I can checkout and test myself I use rustling with students and feel some problems due to mismatch with book chapters/rustlings order.

Hello,

I have tested this locally. But feel free to double check :)

**Comment by AbdouSeck** [comment]
Hi @JuliaCao, this is a very good idea. While I have not pulled the changes locally yet, I don't see how this would not be better than the current ordering of the exercises.

However, I have noticed that you've got a `merge commit` in your commits. This may have stemmed from a desire to keep your branch up-to-date with the `main` branch. In keeping with linear histories (among other things) in the project, would you mind using a `rebase` instead of a `merge`, please?
You may need to reset back to your own commit before the merge and then rebase from there. Please let me know if you have any questions.

Thanks,
Abdou

**Comment by JuliaCao** [comment]
Hi Abdou,

Thanks for getting back on this. I've just rebased and updated my branch. Please take a look and let me know what you think.

Thanks,
Julia

**Review by AbdouSeck — APPROVED** [review]
@JuliaCao thanks for making the change. @fmoko this should be good to go.

**Comment by fmoko** [comment]
I'm not sure whether this would necessitate a breaking change?

**Comment by AbdouSeck** [comment]
@fmoko none that I could detect. The way the `info.toml` file de-serialized is hidden from the end users and there does not seem to be anything else in the source code relying on the order of the exercises within the code for the `watch` command. But I would not hurt to mention in the change logs that the ordering of the exercises was changed to match the Book's chapters.

Thanks,
Abdou

**Comment by fmoko** [comment]
@AbdouSeck Yup, I think deciding whether it's a breaking change is mostly a semantic thing here, but I'll go with your judgement. Thanks!

**Comment by AbdouSeck** [comment]
@JuliaCao @fmoko I really messed up here. The merged `info.toml` from this PR skipped 5 exercises:
```
 iterators1
collections1
collections2
collections3
collections4
```
This may have happened during `rebase`? Either way, I am happy to bring those back. But I wanted to give @JuliaCao the opportunity to take a stab at it since she's more comfortable with the ordering of the exercises?
Let me know what you think.

Thanks,
Abdou

**Comment by JuliaCao** [comment]
@AbdouSeck Sure. Let me take a look

**Comment by JuliaCao** [comment]
Opened a new PR for a fix here #598

**Comment by AbdouSeck** [comment]
@JuliaCao wonderful. I'll get on it ASAP. Thanks!
